### PR TITLE
CDAP-3063 cli commands for artifacts

### PR DESCRIPTION
**Pull request by albertshau**
Adding commands to list artifacts, get artifact details,
get plugin details, add and delete artifacts, deploy apps
from artifacts, update apps, and filter apps by artifact name
and version.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT2836

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/artifact/DescribeArtifactCommand.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  public String getPattern() {
+    return String.format("describe artifact <%s> <%s> [<%s>]",
+                         ArgumentName.ARTIFACT_NAME, ArgumentName.ARTIFACT_VERSION, ArgumentName.SCOPE);
+  }
+
+  @Override
+  public String getDescription() {
+    return String.format("Shows information about an %s. If no scope is given, the user scope will be used. " +
+      "Includes information about application and plugin classes contained in the artifact.",
+      ElementType.ARTIFACT.getName());
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/artifact/DescribeArtifactCommand.java]: How come in most of the descriptions, you inline 'artifact', but only in two of them, you use string.format to put in 'artifact'?

Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/artifact/DescribeArtifactCommand.java]: no good reason, will change them to put in artifact

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2012-2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java]: nit: 2015

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    ArtifactScope artifactScope = ArtifactScope.valueOf(arguments.get(ArgumentName.SCOPE.toString()).toUpperCase());
+    ArtifactSummary artifact = new ArtifactSummary(artifactName, artifactVersion, artifactScope);
+
+    JsonObject config = new JsonObject();
+    String configPath = arguments.getOptional(ArgumentName.APP_CONFIG_FILE.toString());
+    if (configPath != null) {
+      File configFile = resolver.resolvePathToFile(configPath);
+      FileReader reader = new FileReader(configFile);
+      config = GSON.fromJson(reader, JsonObject.class);
+      reader.close();
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java]: have a try finally and close reader in finally. incase GSON.fromJson fails, we should close the reader

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/UpdateAppCommand.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2012-2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/UpdateAppCommand.java]: nit: 2015

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java (hunk 3 above)
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java]: This comment applies to other places as well.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java (hunk 2 above)
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/CreateAppCommand.java]: please check all the files.

**Comment by sreevatsanraman** [comment]
Minor comments. Fix and merge.

**Pull request merged**